src/parser.js

```diff
@@ -2486,15 +2564,21 @@ export class GenericParser extends Tokenizer {
     this.expect(TokenType.LBRACE);
 
     let properties = [];
+    let rest = null;
     while (!this.match(TokenType.RBRACE)) {
+      if (this.eat(TokenType.ELLIPSIS)) {
+        rest = this.parseBindingIdentifier();
+        break;
+      }
       properties.push(this.parseBindingProperty());
-      if (!this.match(TokenType.RBRACE)) {
+      if (!this.match(TokenType.RBRACE) && !this.match(TokenType.ELLIPSIS)) {
```

Comment:
Same comment on the `&& !this.match(TokenType.ELLIPSIS)` as [above](https://github.com/shapesecurity/shift-parser-js/pull/414/files#r282292413).

Reply:
Added a test